lbrynet/lbrynet_daemon/auth/server.py

```diff
@@ -185,13 +217,23 @@ def expire_session():
         else:
             d = defer.maybeDeferred(function, *args)
 
-        # cancel the response if the connection is broken
-        notify_finish.addErrback(self._response_failed, d)
+        # finished_deferred will fire when the request is finished
+        # this could be because the request is really done, or because the connection dropped
+        # if the connection dropped, cancel the deferred stack
+        # otherwise it'll try writing to a closed request and twisted doesn't like that
+
+        # TODO: don't trap RuntimeError, which is presently done to handle deferredLists that
+        # won't peacefully cancel, namely get_lbry_files
+
+        finished_deferred.addBoth(self._handle_dropped_request, d, function_name)
```

Comment:
The addBoth should be (and has been updated to) addErrback.

>Is there a situation in which finished_deferred will fire a successful callback without d having been called?

There shouldn't be, d being called means request.finish() has been called. It seems like there's a  potential race condition if finished_deferred errbacks though: will d be cancelled before it calls request.finish()? If not, then there will be a RuntimeError instead of a ConnectionLos error. Race conditions aside, this also happens if d can't be successfully cancelled, as is the case with the deferredList returned by Daemon.get_lbry_files - but fixing that can go in a separate pr.